src/v/datalake/record_multiplexer.cc

```diff
@@ -34,11 +46,61 @@ record_multiplexer::operator()(model::record_batch batch) {
         auto record = it.next();
         iobuf key = record.release_key();
         iobuf val = record.release_value();
-        // *1000: Redpanda timestamps are milliseconds. Iceberg uses
-        // microseconds.
-        int64_t timestamp = (first_timestamp + record.timestamp_delta()) * 1000;
+        auto timestamp = model::timestamp{
+          first_timestamp + record.timestamp_delta()};
         kafka::offset offset{batch.base_offset()() + record.offset_delta()};
-        int64_t estimated_size = key.size_bytes() + val.size_bytes() + 16;
+        int64_t estimated_size = key.size_bytes() + val.size_bytes();
+
+        auto val_type_res = co_await _type_resolver.resolve_buf_type(
+          std::move(val));
+        if (val_type_res.has_error()) {
+            switch (val_type_res.error()) {
+            case type_resolver::errc::bad_input:
+            case type_resolver::errc::translation_error:
+                // TODO: metric + use binary data?
+            case type_resolver::errc::registry_error:
+                _error = data_writer_error::parquet_conversion_error;
```

Comment:
nit: add a log line with the error type.

Reply:
Done